pkg/kritis/cryptolib/pgp.go

```diff
@@ -51,15 +52,72 @@ func (v pgpVerifierImpl) verifyPgp(signature, publicKey []byte) ([]byte, error)
 	// This will call PublicKey.VerifySignature for the keys in the keyring.
 	payload, err := ioutil.ReadAll(messageDetails.UnverifiedBody)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not verify armor signature")
+		return nil, errors.Wrap(err, "error reading message contents")
 	}
 
-	// Make sure after reading the UnverifiedBody above, there is no signature error.
+	// Make sure after reading the UnverifiedBody above that the Signature
+	// exists and there is no SignatureError.
 	if messageDetails.SignatureError != nil {
-		return nil, errors.Wrap(messageDetails.SignatureError, "bad signature found")
+		return nil, errors.Wrap(messageDetails.SignatureError, "failed to validate: signature error")
 	}
 	if messageDetails.Signature == nil {
-		return nil, errors.New("no signature found for given key")
+		return nil, fmt.Errorf("failed to validate: signature missing")
 	}
 	return payload, nil
 }
+
+type pgpSigner struct {
+	privateEntity *openpgp.Entity
+	publicKeyID   string
+}
+
+// NewPgpSigner creates a Signer interface for PGP Attestations. `privateKey`
+// contains the ASCII-armored private key.
+func NewPgpSigner(privateKey []byte) (Signer, error) {
+	keyring, err := openpgp.ReadArmoredKeyRing(bytes.NewReader(privateKey))
+	if err != nil {
+		return nil, errors.Wrap(err, "error reading armored private key")
+	}
+	if len(keyring) != 1 {
+		return nil, fmt.Errorf("expected 1 key in keyring, got %d", len(keyring))
+	}
+	privateEntity := keyring[0]
+	return &pgpSigner{
+		privateEntity: privateEntity,
+		publicKeyID:   fmt.Sprintf("%X", privateEntity.PrimaryKey.Fingerprint),
+	}, nil
+}
+
+// CreateAttestation creates a signed PGP Attestation. The Attestation's
+// publicKeyID will be derived from the private key. See Signer for more
+// details.
+func (s *pgpSigner) CreateAttestation(payload []byte) (*Attestation, error) {
+	// Create a buffer to store the signature
+	signature := bytes.Buffer{}
+
+	// Armor-encode the signature before writing to the buffer
+	armorBuffer, err := armor.Encode(&signature, openpgp.SignatureType, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating armor buffer")
+	}
+
+	armorWriter, err := openpgp.Sign(armorBuffer, s.privateEntity, nil, nil)
```

Comment:
And maybe rename this to reflect what it does (signs).  signatureWriter?

Reply:
Done.